core/src/main/java/org/kiwix/kiwixmobile/core/error/ErrorActivity.java

```diff
@@ -112,16 +116,12 @@ protected void onCreate(Bundle savedInstanceState) {
       String body = getBody();
 
       if (allowLogsCheckbox.isChecked()) {
-        File appDirectory = new File(Environment.getExternalStorageDirectory() + "/Kiwix");
-        File logFile = new File(appDirectory, "logcat.txt");
-        if (logFile.exists()) {
-          Uri path =
-            FileProvider.getUriForFile(this,
-              getApplicationContext().getPackageName() + ".fileprovider",
-              logFile);
-          emailIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
-          emailIntent.putExtra(Intent.EXTRA_STREAM, path);
-        }
+        File file = fileLogger.writeLogFile(this);
+        Log.d("SEARCHING.....", "onCreate: Searching @ "+ "org.kiwix.kiwixmobile"+ ".fileProvider"+ file);
+        Uri path = getUriForFile(this,
+          "org.kiwix.kiwixmobile.fileprovider", file); // Please note the lower-case here of fileprovider
```

Comment:
Should this be fit into a single line?
PS: Lint allows it to be fit in a single line

Reply:
if it can be a single line then it should be a single line, we can probably also get rid of the comment